third_party/move/move-vm/types/src/values/values_impl.rs

```diff
@@ -704,12 +812,117 @@ impl IndexedRef {
         };
         Ok(res)
     }
+
+    fn compare(&self, other: &Self) -> PartialVMResult<Ordering> {
+        use Container::*;
+
+        let res = match (
```

Comment:
Was wondering if the logic would be easier if you do a read_ref on both ends and compare?

Reply:
not sure, why equals doesn't do that? maybe perf?